### DISCUSSION TITLE
Is there any way to get latest deployment identified by projectId and environmentId, without activityID?

### DISCUSSION THREAD

**Discussion by DKobayashi-ykgw** [d365collaborative/d365fo.tools#520] [Q&A]
Hi,
Is there any way to get latest deployment identified by projectId and environmentId, without activityID?
I want to get AssetId where succeed deployment, identified by projectId and environmentId.

**Comment by Splaxi** [comment]
Hi.

From what I know, none of these details are available at the moment. I have been chatting with @richardsondev about stuff like this.

So - just to make sure I understand it correctly:
Your pipeline uploads a package to LCS.
Your pipeline deploys the package to an environment.

Now you want to query the environment, and get the latest deployment, with success, and obtain the AssetId. Is that correct?

With that, you would be able to deploy the same AssetId to another environment, without having to look it up.

**Reply by DKobayashi-ykgw** [reply]
Hi Motz Jensen,

Thanks for your reply!
I want to describe my scenario why I want to get Asset ID.

We have multiple Sandbox environment and that stage is a little bit different. First sandbox is used for test by partner, second sandbox is used by customer. The timing of deployment to each sandboxes are not same. After pass partner test, then deploy to customer sandbox.
Therefore, I want to get latest Asset ID which deployed to partner sandbox.

**Reply by richardsondev** [reply]
Hello, I am from the Microsoft Dynamics LCS engineering team. 

I am currently working on the three below endpoints internally:
1. List environments in project with a project id
2. Get environment status with an environment id + project id
3. Get environment change history with an environment id + project id

This work isn't officially on the LCS roadmap but I am working on them when I can. I hope to have these LCS APIs available (with D365FO.tools methods) sometime in the next 1 - 1.5 months.

I believe the third endpoint will give you the information you need (asset id deployed last in change history).

Regards,
Billy Richardson

**Reply by Splaxi** [reply]
So if you design your pipeline with multiple stages, and were to based the name of the asset with the build name, you can solve it todages.

We have pre-deployment approvals for each stages, to block deployments until we are ready. 

This cmdlet can list and filter all assets based on name or filename, in any stage.

https://github.com/d365collaborative/d365fo.tools/blob/master/docs/Get-D365LcsAssetFile.md

**Reply by DKobayashi-ykgw** [reply]
Thank you Billy, Mötz,
I'm highly appreciated if new LCS API feature would released, it's just I want!
I'll try to use pre-deployment approvals until new feature of LCSAPI release.

Best regards,
Daisuke
